src/sound/WAVFile.as

```diff
@@ -60,14 +60,22 @@ public class WAVFile {
 		// read format chunk
 		var formatChunk:ByteArray = extractChunk('fmt ', waveData);
 		if (formatChunk.length < 16) throw Error("WAVFile: format chunk is too small");
-		var encoding:int = formatChunk.readShort();
+		var encoding:uint = formatChunk.readUnsignedShort();
 
 		result.encoding = encoding;
-		result.channels = formatChunk.readShort();
-		result.samplesPerSecond = formatChunk.readInt();
-		result.bytesPerSecond = formatChunk.readInt();
-		result.blockAlignment = formatChunk.readShort();
-		result.bitsPerSample = formatChunk.readShort();
+		result.channels = formatChunk.readUnsignedShort();
+		result.samplesPerSecond = formatChunk.readUnsignedInt();
+		result.bytesPerSecond = formatChunk.readUnsignedInt();
+		result.blockAlignment = formatChunk.readUnsignedShort();
+		result.bitsPerSample = formatChunk.readUnsignedShort();
+		if (formatChunk.length >= 18 && encoding == 0xFFFE) {
+			var extensionSize:uint = formatChunk.readUnsignedShort();
+			if (extensionSize == 22) {
+				result.validBitsPerSample = formatChunk.readUnsignedShort();
+				result.channelMask = formatChunk.readUnsignedInt();
+				result.encoding = encoding = formatChunk.readUnsignedShort();
```

Comment:
From the spec I linked to: "The first two bytes of the GUID form the sub-code specifying the data format code, e.g. WAVE_FORMAT_PCM. The remaining 14 bytes contain a fixed string, \x00\x00\x00\x00\x10\x00\x80\x00\x00\xAA\x00\x38\x9B\x71."